datafusion/physical-expr/src/aggregate/groups_accumulator/adapter.rs

```diff
@@ -342,6 +374,50 @@ impl GroupsAccumulator for GroupsAccumulatorAdapter {
     fn size(&self) -> usize {
         self.allocation_bytes
     }
+
+    fn convert_to_state(
```

Comment:
thanks @Rachelint 

This might be interesting to you: https://github.com/apache/datafusion/pull/11825

Reply:
> thanks @Rachelint
> 
> This might be interesting to you: #11825

Seems interesting, learning this pr (still not so familiar with arrow ops).

Reply:
Yeah, it takes some getting used to thinking in terms of `Array`s and masks, etc

Reply:
Sorry for delay, fixed.

Reply:
Found the quick way using `filtered_null_mask` + `set_nulls` will just set the filtered row to be null, but now change the row num.
```rust
//  `filtered_null_mask` + `set_nulls`
  left: PrimitiveArray<Int32>
[
  null,
  null,
  6,
  null,
  null,
  null,
]

 // `compute::filter`
 right: PrimitiveArray<Int32>
[
  null,
  6,
]
```
Maybe this will make difference to the correctness for some accumulators? For example, an udf count which will thinks a null row as 1?